### PR TITLE
SNRAY-209: Export ECL labelservice

### PR DESCRIPTION
**Pull request by molnarlaura**

### CODE REVIEW

File: snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelRestInput.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+import java.util.List;
+
+/**
+ * @since 7.17.0
+ */
+public class SnomedExpressionLabelRestInput {
+	
+	private List<String> expressions;
+	private String descriptionType;
```
Comment (cmark) [inline snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelRestInput.java:30]: Set default value to the `FSN` key.

File: snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelRestInput.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+package com.b2international.snowowl.snomed.core.rest;
+
+import java.util.List;
+
+/**
+ * @since 7.17.0
+ */
+public class SnomedExpressionLabelRestInput {
+	
+	private List<String> expressions;
```
Comment (cmark) [inline snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelRestInput.java:28]: Annotate fields with `ApiParam` so the REST UI will be able to pick it up and display it.

File: snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelRestInput.java (hunk 1 above)
Comment (cmark) [reply snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelRestInput.java:30]: Also, include possible values `fsn, pt` somewhere in the `ApiParam` documentation.

File: snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelService.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import com.b2international.snowowl.snomed.core.ql.Expressions;
+import com.b2international.snowowl.snomed.datastore.request.SnomedRequests;
+
+import io.swagger.annotations.*;
+
+/**
+ * @since 7.17.0
+ */
+@Api(value = "label", description="Label", tags = { "label" })
+@Controller
```
Comment (cmark) [inline snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelService.java]: Use `RestController` and remove the `ResponseBody` annotations from method return types.

File: snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelService.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+@RequestMapping(value = "/{path:**}/label") 
+public class SnomedExpressionLabelService extends AbstractSnomedRestService {
+	
+	@ApiOperation(
+			value="Retrieve ECL Labels", 
+			notes="Returns a collection resource containing ECL expressions."
+		)
+	@ApiResponses({
+		@ApiResponse(code = 200, message = "OK", response = Expressions.class),
+		@ApiResponse(code = 400, message = "Invalid search config", response = RestApiError.class),
```
Comment (cmark) [inline snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelService.java]: This is not a search endpoint. Use `Bad Request` if you don't have better alternative.

File: snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelService.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+		@ApiResponse(code = 404, message = "Label not found", response = RestApiError.class)
```
Comment (cmark) [inline snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelService.java]: `404` will be returned when the service does not find the corresponding Path expression, just `Not Found` is enough.

File: snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelService.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+	})
+	@GetMapping(consumes = { AbstractRestService.JSON_MEDIA_TYPE })
+	public @ResponseBody Promise<Expressions> getLabel(			
+			@RequestBody
+			final SnomedExpressionLabelRestInput body,
+			
+			@ApiParam(value = "The branch path", required = true)
```
Comment (cmark) [inline snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelService.java:48]: Move this above the body argument to match the other endpoints.

File: snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelService.java [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+import com.b2international.snowowl.snomed.datastore.request.SnomedRequests;
+
+import io.swagger.annotations.*;
+
+/**
+ * @since 7.17.0
+ */
+@Api(value = "label", description="Label", tags = { "label" })
+@Controller
+@RequestMapping(value = "/{path:**}/label")
```
Comment (cmark) [inline snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelService.java]: I would prefer something like `label-expressions` or something more specific than just `label`.

**Review by cmark — CHANGES_REQUESTED** [review]

File: snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelRestInput.java (hunk 1 above)
Comment (molnarlaura) [reply snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelRestInput.java:30]: Done

File: snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelRestInput.java (hunk 2 above)
Comment (molnarlaura) [reply snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelRestInput.java:28]: Done

File: snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelService.java (hunk 3 above)
Comment (molnarlaura) [reply snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelService.java]: Done

File: snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelService.java (hunk 7 above)
Comment (molnarlaura) [reply snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelService.java]: Done

File: snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelService.java (hunk 6 above)
Comment (molnarlaura) [reply snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelService.java:48]: Done

File: snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelService.java (hunk 4 above)
Comment (molnarlaura) [reply snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelService.java]: Done

File: snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelService.java (hunk 5 above)
Comment (molnarlaura) [reply snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelService.java]: Done

File: snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/AllSnomedApiTests.java [hunk 8]
```diff
@@ -112,6 +112,7 @@
 	// Performance test cases, should be the last tests to perform
 	SnomedConceptCreatePerformanceTest.class,
 	SnomedMergePerformanceTest.class,
+	SnomedExpressionLabelTest.class
```
Comment (cmark) [inline snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/AllSnomedApiTests.java]: Please note the comment above. Performance benchmark tests should be the last tests to execute, please move this up in the execution order, could be next to the component API tests.

File: snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelRestInput.java [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+import io.swagger.annotations.ApiParam;
+
+/**
+ * @since 7.17.0
+ */
+public class SnomedExpressionLabelRestInput {
+	
+	@ApiParam
+	private List<String> expressions;
+	@ApiParam(defaultValue = "fsn", example = "{fsn, pt}")
```
Comment (cmark) [inline snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelRestInput.java]: How about `notes` or `description`? Not sure if `example` will work as expected.

**Review by cmark — CHANGES_REQUESTED** [review]

File: snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/AllSnomedApiTests.java (hunk 8 above)
Comment (molnarlaura) [reply snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/AllSnomedApiTests.java]: Done

File: snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelRestInput.java (hunk 9 above)
Comment (molnarlaura) [reply snomed/com.b2international.snowowl.snomed.core.rest/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelRestInput.java]: Done

File: snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelTest.java [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+		String conceptId = createNewConcept(branchPath);
+
+		String shortName = "SNOMEDCT-Test";
+		createCodeSystem(branchPath, shortName).statusCode(201);
+		
+		List<String> expressionLabels = getExpressionLabels(branchPath.getPath(), List.of(conceptId));
+		
+		SnomedConcept concept = getConcept(conceptId, "term()");
+		
+		expressionLabels.stream().findFirst().ifPresent(label -> label.equals(conceptId + " |" + concept.getFsn() + "|" ));
```
Comment (cmark) [inline snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelTest.java]: ⚠️ There is no assert in this test case 😢.

**Review by cmark — CHANGES_REQUESTED** [review]

File: snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelTest.java (hunk 10 above)
Comment (molnarlaura) [reply snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/SnomedExpressionLabelTest.java]: Fixed it 😓

**Review by cmark — APPROVED** [review]
LGTM!

**Pull request merged**